File: crabzilla-vertx/src/main/java/io/github/crabzilla/stack/ProjectionData.java
package io.github.crabzilla.stack;

import lombok.Value;
import io.github.crabzilla.model.DomainEvent;

import java.io.Serializable;
import java.util.List;
import java.util.UUID;

@Value
public class ProjectionData implements Serializable {

  UUID uowId;
  Long uowSequence;
  String targetId;
  List<DomainEvent> events;

}

File: crabzilla-core/src/main/java/io/github/crabzilla/ProjectionData.kt
package io.github.crabzilla

import java.util.*

data class ProjectionData(val uowId: UUID, val uowSequence: Int, val entityId: Int,
                          val events: List<Pair<String, DomainEvent>>) {
  companion object {
    @JvmStatic
    fun fromUnitOfWork(uowSequence: Int, uow: UnitOfWork) : ProjectionData {
      return ProjectionData(uow.unitOfWorkId, uowSequence, uow.entityId, uow.events)
    }
  }
}
